src/main/scala/com/microsoft/ml/spark/vw/VowpalWabbitBase.scala

```diff
@@ -193,51 +221,87 @@ trait VowpalWabbitBase extends Wrappable
     val featureColIndices = generateNamespaceInfos(df.schema)
 
     def trainIteration(inputRows: Iterator[Row],
-                       localInitialModel: Option[Array[Byte]]): Iterator[Option[Array[Byte]]] = {
+                       localInitialModel: Option[Array[Byte]]): Iterator[TrainingResult] = {
 
       // construct command line arguments
       val args = buildCommandLineArguments(vwArgs, contextArgs)
 
+      val totalTime = new StopWatch
```

Comment:
maybe call these timers an rename the stopwatch class to Timer

Reply:
not sure if the comment is motivated by Java/Scala notion. In C# timers are recurring event triggers and StopWatch (https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch?view=netframework-4.8) are used to measure time...

Reply:
Okay ill defer to your convention!